src/reporting/iterators/IteratorUtil.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { PagedResponseLinks } from "../interfaces/Links";
+
+export interface Collection<TEntity> {
+  values: Array<TEntity>;
+  _links: PagedResponseLinks;
+}
+
+export interface EntityCollectionPage<TEntity> {
+  /** Current page entities. */
+  entities: TEntity[];
+  /** Function to retrieve the next page of the entities. If `undefined` the current page is last. */
+  next?: () => Promise<EntityCollectionPage<TEntity>>;
+}
+
+export type EntityPageQueryFunc<TEntity> = () => Promise<EntityCollectionPage<TEntity>>;
+
+export async function* flatten<TEntity>(pagedIterator: AsyncIterableIterator<TEntity[]>): AsyncIterableIterator<TEntity> {
+  for await (const entityChunk of pagedIterator) {
+    for (const entity of entityChunk)
+      {yield entity;}
+  }
+}
+
+export async function getEntityCollectionPage<TEntity>(
+  nextUrl: string,
+  requestOptions: RequestInit,
```

Comment:
`requestOptions` are not changed or read in this method, so I think it could be baked into the `getNextBatch` function.

Reply:
removed requestOptions as a variable and made a const request variable which is not a parameter of the getEntityCollectionPage or getNextBatch functions but is used by them.